### DISCUSSION TITLE
Change regularization for crosshole traveltime tomograpy

### DISCUSSION THREAD

**Discussion by sminize** [gimli-org/pyGIMLi#355] [Q&A]
Hi everyone,
is there a way to change the regularization for traveltime crosshole tomography? (see the example here: https://www.pygimli.org/_examples_auto/2_seismics/plot_02_crosshole_tomography.html#sphx-glr-examples-auto-2-seismics-plot-02-crosshole-tomography-py)

I want to change the regularization to the total variation regularization in order to get sharp edges in the inversion results.

Thank you very much for your help and a happy new year to everyone. :)

Best regards,

Simon

**Comment by halbmy** [comment]
Generally, you should be able to create your own regularization matrix, but we would need a better API and documentation for it. Furthermore, as TV depends on the model itself, it is even a bit more complicating. We consider this when overhauling the inversion.

In the meanwhile, you can use `mgr.invert(blockyModel=True)` which also enhances contrasts.

**Reply by sminize** [reply]
Thank you very much for your fast reply and the tip using `mgr.invert(blockyModel=True)`. Which regularization is used in this case?

Also thank you for considering the implementation of the TV regularization. 

All the best,
Simon Schmid
